### DISCUSSION TITLE
IoTDB-UI officially open source

### DISCUSSION THREAD

**Discussion by limeng32** [apache/iotdb#5094] [Show and tell]
Our company's IoTDB management system IoTDB-UI now officially open source, the front-end address is https://github.com/limeng32/iotdbui-front, the back-end address is https://github.com/limeng32/iotdbui-back, and the demo address is http://115.28.134.232:8800/. We will maintain this project for a long time, and I hope you would not hesitate to put forward your own needs and advices.

**Comment by limeng32** [comment]
Ver-0.12.3 released!
Contents of this update:

1. Metadata uses streaming loading, which can manage massive sg, timeseries, privileges;

2. The data in the query result set can be directly exported as an excel file;

3. Time in the query result set can be switched between number and UTC time;

4. 'Time' can be used for data locate in the query result set;

The next expected update content:

IoTDB-UI user management module;

Enjoy it.

**Comment by limeng32** [comment]
Ver-0.12.4 released!
Contents of this update:

1. An email-based account management system;

2. The sql input box can now comment, execute and format sql via shortcut keys;

3. Improve page performance and fix known bugs;

The demo address is http://115.28.134.232:8800/ , if you have any requirements and suggestions, you can write here or https://issues.apache.org/jira/projects/IOTDB/issues/IOTDB-2535;

Enjoy it.

**Comment by jixuan1989** [comment]
Hi, I suggest to post this info to the mailing list and Slack to get more attentions.

**Comment by limeng32** [comment]
+1, very honoured



***@***.***
 
From: Xiangdong Huang
Date: 2022-04-11 21:03
To: apache/iotdb
CC: limeng; Author
Subject: Re: [apache/iotdb] IoTDB-UI officially open source (Discussion #5094)
Hi, I suggest to post this info to the mailing list and Slack to get more attentions.

**Comment by limeng32** [comment]
IoTDB-UI docker-0.12.4 released!

For the B/S architecture is a bit complicated to deploy, every new version of IoTDB-UI from now on will add a docker image version, making deployment a breeze.

Image name:

    limeng32/iotdbui

The easiest way to use it is:

    1. Use:
            docker pull limeng32/iotdbui 
        or 
            docker pull limeng32/iotdbui:0.12.4 
        to get the latest image
    2. Use:
            docker run -p 80:80 -it --rm limeng32/iotdbui 
        to get started right away

The build address of the image limeng32/iotdbui is as follows (you can refer to README.md for more complex settings):

    https://github.com/limeng32/iotdbui-back/tree/docker

0.12.4 Main content:

    1. Compatible with IoTDB-0.13.0 syntax
    2. Add email-based account management module
    3. Add some hotkeys related to the query
    4. Fixed known bugs

0.12.5 Expected content:
    
    Graphical query results
